### ISSUE TITLE
Fix the app data/cache management

### ISSUE THREAD

**Issue by nvllz** [Demizo/Daily_You#24]
I just found that adding new entries with images increases the app's data and cache size, but deleting entries doesn't reduce the space used by the app.

**Comment by Demizo** [comment]
Currently the app does not delete images when you remove an entry. I don't know if this behavior should change since removing an entry or removing an image from an entry doesn't necessarily mean you want the image deleted. Perhaps a extra confirmation to delete the image, or maybe it only deletes images automatically if they are stored in the app data and not external storage?

Maybe I am missing the mark on this however. Thoughts?

**Comment by Demizo** [comment]
The image cache is handled by flutter. Its for faster load times since the images are cached at a lower resolution for the cards. The reason the user storage doesn't go down is because the images are still in the app data folder even after their entry is removed. I have this behavior so if you take a photo (and you set the image folder to your camera roll), removing an entry won't take that photo out of your camera roll. With that said this behavior isn't very intuitive and doesn't make sense for images in the app itself.

**Comment by Demizo** [comment]
As for adding gallery images, this occurs because the app is copying the image into its app data. Entries only store image names not full file paths, so all images must be in the same folder (the app data folder by default). This won't change since it is core to flexibility. If you set the image folder to the gallery and add an image from the gallery it shouldn't create a copy (I will double check that because it might right now). Either way it won't increase the app data size.

**Comment by Demizo** [comment]
@nvllz Ah, you were right! There was cache growth I hadn't accounted for. This will be fixed in the next update alongside a confirmation dialog for whether you want to delete images associated with entries or just remove the image from the entry (a similar popup will dialog will apply when deleting entries with images).

Thank you for pointing this out! 🎉

**Comment by Demizo** [comment]
Yeah, everything you say is true. There is a very unfortunate grey area surrounding the images being separate from logs that need to be fleshed out. I agree that the current confirmation dialogs are a bit, well lets just say rough. I will give some thought about how to improve this, the best option may be to just delete the photos. At the very least if the photo is in the app folder it should always be deleted.

I greatly appreciate these suggestions and will reopen the issue for the next release.

**Comment by Demizo** [comment]
@nvllz I gave it some thought. Two layers of confirmation dialogs was confusing and annoying. Just deleting the photo makes the most sense in the vast majority of use cases. It may be worth still adding an option to not delete images in the settings. With that said the biggest confusion point I foresee now is that if I add an image from my gallery and I delete the log or photo, it won't delete the gallery image even though the dialog makes it seem like it might. 

Thanks again for the insight and feedback!
